media/server/gstplayer/source/GstGenericPlayer.cpp

```diff
@@ -382,6 +382,75 @@ bool GstGenericPlayer::getPosition(std::int64_t &position)
     return true;
 }
 
+GstElement *GstGenericPlayer::getSink(const MediaSourceType &mediaSourceType)
+{
+    const char *kSinkName{nullptr};
+    GstElement *sink{nullptr};
+    switch (mediaSourceType)
+    {
+    case MediaSourceType::AUDIO:
+        kSinkName = "audio-sink";
+        break;
+    case MediaSourceType::VIDEO:
+        kSinkName = "video-sink";
+        break;
+    default:
+        break;
+    }
+    if (!kSinkName)
+    {
+        RIALTO_SERVER_LOG_WARN("mediaSourceType not supported %d", static_cast<int>(mediaSourceType));
+    }
+    else
+    {
+        m_glibWrapper->gObjectGet(m_context.pipeline, kSinkName, &sink, nullptr);
+        if (!sink)
+        {
+            RIALTO_SERVER_LOG_WARN("%s could not be obtained", kSinkName);
+        }
+    }
+    return sink;
+}
+
+bool GstGenericPlayer::setImmediateOutput(const MediaSourceType &mediaSourceType, bool immediateOutputParam)
+{
+    bool returnValue{false};
+    GstElement *sink = getSink(mediaSourceType);
+    if (sink)
+    {
+        // For AutoVideoSink we use properties on the child sink
+        if (MediaSourceType::VIDEO == mediaSourceType)
+            sink = getSinkChildIfAutoVideoSink(sink);
+
+        gboolean immediateOutput(immediateOutputParam);
+        m_glibWrapper->gObjectSet(sink, "immediate-output", immediateOutput, nullptr);
+        returnValue = true;
```

Comment:
We shouldn't make any changes to objects in the pipeline in the main thread, create a new task